client/pkg/transport/listener_test.go

```diff
@@ -573,3 +578,153 @@ func TestSocktOptsEmpty(t *testing.T) {
 		}
 	}
 }
+
+// TestNewListenerWithACRLFile tests when a revocation list is present.
+func TestNewListenerWithACRLFile(t *testing.T) {
+	clientTLSInfo, err := createSelfCertEx(t, "127.0.0.1", x509.ExtKeyUsageClientAuth)
+	if err != nil {
+		t.Fatalf("unable to create client cert: %v", err)
+	}
+	loaded, err := os.ReadFile(clientTLSInfo.CertFile)
+	if err != nil {
+		t.Fatalf("unable to read client cert: %v", err)
+	}
+	block, _ := pem.Decode(loaded)
+	clientCert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		t.Fatalf("unable to parse client cert: %v", err)
+	}
+
+	tests := map[string]struct {
+		overrideCRLFileName    bool
+		expectHanshakeError    bool
+		crlFile                string
```

Comment:
If I understood it correctly, the only reason to add `crlFile` is to verify that a bad crlFile path/name will fail the test. It is out of scope and we don't have to cover it. I suggest to remove it to simplify the test case.
```suggestion
```

Reply:
Hi @ahrtr, I was trying to have full coverage of the code. This was testing specifying a CRL File path that doesn't exist (https://github.com/etcd-io/etcd/blob/4e986363a371ad88479cb8f9b248d16ea83abc87/client/pkg/transport/listener_tls.go#L172-L174). I can remove that test case if you feel is not relevant.

Reply:
The case is a little over complicated. It took me more than 10 minutes to read the test case, but still did not fully understand it. It means it would be a little painful for other contributors to update your test case in future. So It would be great if you can simplify the case. 

Yes, It also seems not quite relevant.

Reply:
Sounds good! I'll remove it :scissors: